### DISCUSSION TITLE
Local MQTT/Pub/Sub - to notify component crashes

### DISCUSSION THREAD

**Discussion by chandar1** [aws/aws-iot-device-sdk-cpp-v2#722] [Q&A]
I have 2 apps running as green grass components. 

when one of the apps crashes for some reason, the other app/component should be notified via local pub/sub mechanism - but the crashing app cannot "publish" its status here. 
Is there a way that Green Grass nucleus notifies such status changes, crashes of components, to other running components? 
Need to use pub-sub model here and NOT direct http rest api calls. 

How to achieve this in aws green grass?

Any help is appreciated! 
Thanks.

**Comment by jmklix** [comment]
~~You will want to use [IoT lifecycle events](https://docs.aws.amazon.com/iot/latest/developerguide/life-cycle-events.html). If you subscribe to this topic: `$aws/events/presence/disconnected/clientId`, you will get notified when a specific clientId is disconnected.~~

**Reply by bretambrose** [reply]
> You will want to use [IoT lifecycle events](https://docs.aws.amazon.com/iot/latest/developerguide/life-cycle-events.html). If you subscribe to this topic: `$aws/events/presence/disconnected/clientId`, you will get notified when a specific clientId is disconnected.

I don't think that applies to local pub/sub on GG.

**Reply by jmklix** [reply]
I believe you can use lifecycle events, as Michael Dombrowski suggested it [here](https://repost.aws/questions/QUSB7gx6YJTq2RdjiUmeTq1Q/what-is-the-best-way-to-set-an-iot-last-will-message-on-greengrass-devices): 

> 
> Hi,
> 
> You cannot configure a will for Greengrass. Why do you want a will to be set? If you'd like to know when Greengrass connects and disconnects you can use IoT lifecycle events for that: https://docs.aws.amazon.com/iot/latest/developerguide/life-cycle-events.html
> 
> Cheers,
> 
> Michael

**Reply by bretambrose** [reply]
> I believe you can use lifecycle events, as Michael Dombrowski suggested it [here](https://repost.aws/questions/QUSB7gx6YJTq2RdjiUmeTq1Q/what-is-the-best-way-to-set-an-iot-last-will-message-on-greengrass-devices):
> 
> > Hi,
> > You cannot configure a will for Greengrass. Why do you want a will to be set? If you'd like to know when Greengrass connects and disconnects you can use IoT lifecycle events for that: https://docs.aws.amazon.com/iot/latest/developerguide/life-cycle-events.html
> > Cheers,
> > Michael

They're not asking for GG connectivity they're wondering about component stability.

**Reply by jmklix** [reply]
Thanks @bretambrose for pointing this out, my above suggestion won't work for your situation. Still looking for a solution

**Comment by joysl** [comment]
you can subscribe to component events: https://docs.aws.amazon.com/greengrass/v2/developerguide/ipc-component-lifecycle.html#ipc-operation-subscribetocomponentupdates

**Comment by jmklix** [comment]
The only local solution is to use [GetComponentDetails](https://docs.aws.amazon.com/greengrass/v2/developerguide/ipc-local-deployments-components.html#ipc-operation-getcomponentdetails) to get the status of each component. This might not fix your exact use case because it's something that you have check yourself. Please let me know if you have any questions about this
